src/main/java/com/google/gcloud/storage/StorageServiceImpl.java

```diff
@@ -120,25 +124,41 @@ public StorageObject call() {
   public Bucket get(String bucket, BucketSourceOption... options) {
     final com.google.api.services.storage.model.Bucket bucketPb = Bucket.of(bucket).toPb();
     final Map<StorageRpc.Option, ?> optionsMap = optionMap(options);
-    return Bucket.fromPb(runWithRetries(
+    com.google.api.services.storage.model.Bucket answer = runWithRetries(
         new Callable<com.google.api.services.storage.model.Bucket>() {
           @Override
           public com.google.api.services.storage.model.Bucket call() {
-            return storageRpc.get(bucketPb, optionsMap);
+            try {
+              return storageRpc.get(bucketPb, optionsMap);
+            } catch (StorageServiceException ex) {
+              if (ex.code() == 404) {
+                return null;
+              }
+              throw ex;
+            }
           }
-        }, retryParams, EXCEPTION_HANDLER));
+        }, retryParams, EXCEPTION_HANDLER);
+    return answer == null ? null : Bucket.fromPb(answer);
   }
 
   @Override
   public Blob get(String bucket, String blob, BlobSourceOption... options) {
     final StorageObject storedObject = Blob.of(bucket, blob).toPb();
     final Map<StorageRpc.Option, ?> optionsMap = optionMap(options);
-    return Blob.fromPb(runWithRetries(new Callable<StorageObject>() {
+    StorageObject storageObject = runWithRetries(new Callable<StorageObject>() {
       @Override
       public StorageObject call() {
-        return storageRpc.get(storedObject, optionsMap);
+        try {
+          return storageRpc.get(storedObject, optionsMap);
+        } catch (StorageServiceException ex) {
+          if (ex.code() == 404) {
```

Comment:
Honestly, I used to be once in the camp that creates a constant for every "magic number" but at some point and for some cases I found it less readable and preferred the embedded value with
optionally a comment. Especially when there is only one case of it. However, this is not the case here and I could use an already pre-defined constant, so I will send a CL to fix it.